src/TestFramework/Coverage/TestLocations.php

```diff
@@ -39,48 +39,41 @@
 
 /**
  * @internal
- *
- * ```
- * 'byMethod' => [
- *      'mutate' => MethodLocationData::['startLine' => 12, 'endLine' => 16],
- *      ...
- * ],
- * 'byLine' => [
- *     22 => [
- *         CoverageLineData::[
- *             'testMethod' => '\A\B\C::test_it_works',
- *             'testFilePath' => '/path/to/A/B/C.php',
- *             'time' => 0.34325,
- *         ],
- *         ...
- *      ]
- *  ]
- * ```
  */
 final class TestLocations
 {
+    private $byLine;
+    private $byMethod;
+
     /**
-     * @var array<int, array<int, TestLocation>>
+     * @param array<int, array<int, TestLocation>> $testLocationsBySourceLine
+     * @param array<string, SourceMethodRange> $sourceMethodRangeByMethod
      */
-    public $byLine = [];
+    public function __construct(
+        array $testLocationsBySourceLine = [],
+        array $sourceMethodRangeByMethod = []
+    ) {
+        $this->byLine = $testLocationsBySourceLine;
+        $this->byMethod = $sourceMethodRangeByMethod;
+    }
 
     /**
-     * TODO: use a getter "get"
+     * This method needs to be able to return a reference for performance reasons.
```

Comment:
I wonder if there's actual penalty here.

Reply:
Definitely as otherwise you would create a new array reference at the very least here and would need to create a new object upstream to assign the modified one. That said since the content is all objects, their references are kept and not copied, so maybe the performance impact won't be as much.

Might be worth to revisit